src/Interpreters/Session.cpp

```diff
@@ -326,6 +326,12 @@ void Session::authenticate(const Credentials & credentials_, const Poco::Net::So
 #endif
 }
 
+void Session::resetAuthentication()
+{
+    user_id.reset();
+    *prepared_client_info = {};
```

Comment:
`user` should also be reset, not only `user_id`. And it seems there can be some fields in `prepared_client_info` which should not be reset, so I think it's better to reset only authentication-related fields in `prepared_client_info`. And please check if `prepared_client_info` is not null before.

Reply:
Ok